compiler/src/main/scala/org/scalajs/nscplugin/PrepJSInterop.scala

```diff
@@ -377,6 +377,52 @@ abstract class PrepJSInterop[G <: Global with Singleton](val global: G)
             }
           }
 
+        /* Rewrite js.dynamicImport[T](body) into
+         *
+         * runtime.dynamicImport[A](
+         *   new DynamicImportThunk { def apply(): Any = body }
+         * )
+         */
+        case Apply(TypeApply(fun, List(tpeArg)), List(body))
+            if fun.symbol == JSPackage_dynamicImport =>
+          val pos = tree.pos
+
+          assert(currentOwner.isTerm, s"unexpected owner: $currentOwner at $pos")
+
+          val clsSym = currentOwner.newClass(tpnme.ANON_CLASS_NAME, pos)
```

Comment:
Aren't we supposed to append a unique `$i` at this point? Perhaps not, I'm not sure.

Reply:
No. The trees that reach `PrepJSInterop` for anonymous classes do not have an `$i` yet.